### PR TITLE
Watermark search endpoint

### PR DESCRIPTION
**Pull request by om-sharma**
There is a separate task introduced which uploads the data in cloudsearch domain. This task reads the read api database and periodically uploads the difference. There is a separate watermark domain maintained which keeps the serial number of last entry uploaded to the domain.
 This PR was needed to solve the issue mentioned in https://github.com/openregister/indexer/pull/10.

### CODE REVIEW

File: src/main/java/uk/gov/indexer/Application.java [hunk 1]
```diff
@@ -23,15 +24,33 @@ public static void main(String[] args) throws IOException, SQLException, Interru
...
+                        0,
+                        10,
+                        TimeUnit.SECONDS
+                );
+
+                configuration.cloudSearchEndPoint(register).ifPresent(
+                        endPoint -> executorService.scheduleAtFixedRate(
+                                new CloudSearchDataUploadTask(register, endPoint, configuration.cloudSearchWaterMarkEndPoint(register).get(), dbi.onDemand(IndexedEntriesUpdateDAO.class)),
+                                0,
+                                10,
```
Comment (philandstuff) [inline src/main/java/uk/gov/indexer/Application.java]: given search systems have a greater latency anyway, I wonder if we might want this to be a little less frequent. A minute?

File: src/main/java/uk/gov/indexer/dao/DestinationDBUpdateDAO.java [hunk 2]
```diff
@@ -38,24 +40,47 @@ public int lastReadSerialNumber() {
...
-        upsertInCurrentKeysTable(orderedIndexEntry);
+        List<OrderedEntryIndex> orderedEntryIndex = entries.stream().map(Entry::dbEntry).collect(Collectors.toList());
+        indexedEntriesUpdateDAO.writeBatch(orderedEntryIndex);
+        totalRegisterEntriesUpdateDAO.increaseTotalEntriesInRegisterCount(orderedEntryIndex.size());
+        upsertInCurrentKeysTable(registerName, orderedEntryIndex);
     }
 
-    private void upsertInCurrentKeysTable(List<OrderedIndexEntry> orderedIndexEntry) {
-        List<String> allKeys = Lists.transform(orderedIndexEntry, OrderedIndexEntry::getPrimaryKey);
+    private void upsertInCurrentKeysTable(String registerName, List<OrderedEntryIndex> orderedEntryIndex) {
```
Comment (philandstuff) [inline src/main/java/uk/gov/indexer/dao/DestinationDBUpdateDAO.java]: this is now quite a long method

File: src/main/java/uk/gov/indexer/Application.java (hunk 1 above)
Comment (om-sharma) [inline src/main/java/uk/gov/indexer/Application.java]: Discussed in person, 10 seconds are fine as it is a wait time in successive execution of task.

File: src/main/java/uk/gov/indexer/dao/DestinationDBUpdateDAO.java (hunk 2 above)
Comment (om-sharma) [inline src/main/java/uk/gov/indexer/dao/DestinationDBUpdateDAO.java]: Refatcored :+1:

**Pull request merged**